bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -120,13 +127,20 @@ def get_bcl2fastq_version_from_run_parameters(runfolder, config):
         :return the version of bcl2fastq to use.
         """
 
-        meta_data = InteropMetadata(runfolder)
-        model = meta_data.model
+        run_info = Bcl2FastqConfig.runinfo_as_dict(runfolder)
+        instrument_name = run_info["RunInfo"]["Run"]["Instrument"]
 
-        current_config = config
-        version = current_config["machine_type"][model]["bcl2fastq_version"]
+        machine_type_mappings = {"M": "MiSeq",
+                                 "D": "HiSeq 2500",
+                                 "SN": "HiSeq 2000",
+                                 "ST": "HiSeq X",
+                                 "A": "NovaSeq",
+                                 "NS": "NextSeq 500",
+                                 "K": "HiSeq 4000"}
 
-        return version
+        for key, value in machine_type_mappings.items():
+            if instrument_name.startswith(key):
```

Comment:
if future machine type prefixes are superstrings/substrings of existing prefixes, we need to remember to match more carefully

Reply:
Yes. However at the moment they are not, and fingers crossed, Illumina will keep the prefixes unique (or provide some better way of identifying the machine type).